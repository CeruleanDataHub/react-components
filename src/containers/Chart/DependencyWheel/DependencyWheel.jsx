import PropTypes from "prop-types";
import React from "react";

import { Chart } from "../Chart";

const options = {
  accessibility: {
    point: {
      valueDescriptionFormat:
        "{index}. From {point.from} to {point.to}: {point.weight}."
    }
  }
};

export const DependencyWheel = ({ title, xAxis, series, containerProps }) => (
  <Chart
    title={title}
    xAxis={xAxis}
    series={series}
    options={options}
    type="dependencywheel"
    containerProps={containerProps}
  />
);

DependencyWheel.propTypes = {
  ...Chart.propTypes,
  /** Title of the chart */
  title: Chart.propTypes.title,
  /** Values for X axis */
  xAxis: Chart.propTypes.xAxis,
  /** Data to be graphed */
  series: Chart.propTypes.series,
  /** Props to be passed to highcharts, such as styles or classname */
  // eslint-disable-next-line react/forbid-prop-types
  containerProps: PropTypes.object
};

DependencyWheel.defaultProps = {
  title: Chart.defaultProps.title,
  xAxis: Chart.defaultProps.xAxis,
  series: Chart.defaultProps.series,
  containerProps: null
};
