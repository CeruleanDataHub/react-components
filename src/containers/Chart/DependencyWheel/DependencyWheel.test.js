import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { DependencyWheel } from "./DependencyWheel";

describe("Dependency Wheel", () => {
  it("should render chart", () => {
    const component = shallow(<DependencyWheel />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
