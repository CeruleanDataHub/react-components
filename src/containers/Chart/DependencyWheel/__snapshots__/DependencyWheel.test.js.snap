// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dependency Wheel should render chart 1`] = `
<Chart
  containerProps={null}
  options={
    Object {
      "accessibility": Object {
        "point": Object {
          "valueDescriptionFormat": "{index}. From {point.from} to {point.to}: {point.weight}.",
        },
      },
    }
  }
  series={Array []}
  title=""
  type="dependencywheel"
  xAxis={null}
/>
`;

exports[`Dependency Wheel should render chart with data and title 1`] = `
<Chart
  containerProps={null}
  options={
    Object {
      "accessibility": Object {
        "point": Object {
          "valueDescriptionFormat": "{index}. From {point.from} to {point.to}: {point.weight}.",
        },
      },
    }
  }
  series={
    Array [
      Object {
        "data": Array [
          Array [
            "Brazil",
            "Canada",
            5,
          ],
          Array [
            "Brazil",
            "Mexico",
            1,
          ],
          Array [
            "Canada",
            "Brazil",
            1,
          ],
          Array [
            "Canada",
            "Mexico",
            5,
          ],
          Array [
            "Mexico",
            "Brazil",
            1,
          ],
          Array [
            "Mexico",
            "Canada",
            5,
          ],
        ],
        "dataLabels": Object {
          "color": "#333",
          "distance": 10,
          "textPath": Object {
            "attributes": Object {
              "dy": 5,
            },
            "enabled": true,
          },
        },
        "keys": Array [
          "from",
          "to",
          "weight",
        ],
        "name": "Dependency wheel series",
        "size": "95%",
      },
    ]
  }
  title="Dependency wheel"
  type="dependencywheel"
  xAxis={null}
/>
`;
